_P222_DebouncedCounter.ino

```diff
@@ -0,0 +1,243 @@
+#ifdef USES_P222
+//#######################################################################################################
+//#################################### Plugin 222: Pulse  ###############################################
+//#######################################################################################################
+
+#define PLUGIN_222
+#define PLUGIN_ID_222         222
+#define PLUGIN_NAME_222       "Generic - Debounced Counter"
+#define PLUGIN_VALUENAME1_222 "Count"
+#define PLUGIN_VALUENAME2_222 "Total"
+#define PLUGIN_VALUENAME3_222 "Time"
+
+void Plugin_222_pulse_interrupt1() ICACHE_RAM_ATTR;
+void Plugin_222_pulse_interrupt2() ICACHE_RAM_ATTR;
+void Plugin_222_pulse_interrupt3() ICACHE_RAM_ATTR;
+void Plugin_222_pulse_interrupt4() ICACHE_RAM_ATTR;
```

Comment:
I don't see these functions in your code.
Also doesn't it make more sense to have 1 function and just look at the internal registers to fetch all pin states to simply see what's changed and register those as flags that need to be taken care of in the normal loop?
To fetch all pin states, it takes a single read. Compare it with the previous known state is a single XOR operation and then store the difference in a global variable.
If not 0, it means something was changed which must then be handled with in the 50/sec loop.
You may also apply a set mask to only trigger on the pins you're interested in.